### PR TITLE
release notes update with latest features

### PR DESCRIPTION
**Pull request by JustinSGray**
just trying to capture the recent updates

### CODE REVIEW

File: release_notes.txt [hunk 1]
```diff
@@ -1,18 +1,48 @@
----------------------------------------------------------------------------------------------------
+################################################################################################
+Release Notes for OpenMDAO 2.1.0
+December 1, 2017
+
+New Features:
+-------------
+- Configure setup hook allowing changing of solver settings after hierarchy tree is instantiated
```
Comment (kmarsteller) [inline release_notes.txt]: Minor note, but I had been ending all the items with periods in previous notes.

File: release_notes.txt [hunk 2]
```diff
@@ -1,18 +1,48 @@
...
+
+
+API Changes:
+-------------
+- Case Recording options API updated with `.recording_options` attribute on Driver, Solver, and System classes
+
+- `NonlinearRunOnce` changed `NonLinearRunOnce` for consistency (old class deprecated)
+- `types_` argument to `self.metadata.declare` changed to `types`. (old argument deprecated)
+- `types` and `values` arguments to `self.metadata.declare`
+- `get_subsystem` changed to a private method, removed from public api of System
```
Comment (kmarsteller) [inline release_notes.txt]: capitalize API

File: release_notes.txt [hunk 3]
```diff
@@ -1,18 +1,48 @@
----------------------------------------------------------------------------------------------------
+################################################################################################
+Release Notes for OpenMDAO 2.1.0
+December 1, 2017
+
+New Features:
```
Comment (thearn) [inline release_notes.txt]: Might be worth mentioning `RegularGridInterpComp`, which was moved from pycycle.

Comment (kmarsteller) [reply release_notes.txt]: And worth now changing it to the new name.

**Pull request closed without merging**